### ISSUE TITLE
Installing on Windows

### ISSUE THREAD

**Issue by eliemichel** [facebookresearch/pytorch3d#252]
##  🐛 Bugs / Unexpected behaviors

I was not able to import `pytorch3d._C` after installing on Windows using several methods.

## Instructions To Reproduce the Issue:

Setup: Fresh Python 3.8.1 install, Fresh PyTorch 1.5.1 / CUDA 10.2 install, Windows 10

### First attempt, using pip

```
me@home$ pip install "git+https://github.com/facebookresearch/pytorch3d.git"
[...] (see full log bellow)
me@home$ set PATH=%PATH%;E:\Python38\Lib\site-packages\torch\lib
me@home$ python
>>> from pytorch3d import _C
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: DLL load failed while importing _C: The specified module could not be found.
```

Even adding `site-packages\torch\lib` to the PATH is not enough, though according to [Dependencies](https://github.com/lucasg/Dependencies) all the required dll are in located.

### Second attempt, using local clone:

I followed the instructions from [this section](https://github.com/facebookresearch/pytorch3d/blob/master/INSTALL.md#2-install-from-a-local-clone). Code changes were roughly the same for torch 1.5 than for 1.4. The command `pip install -e .` runs correctly in the developper command prompt, running msvc for some time, then installing stuff in site-packages. But I still get the same error:

```
ImportError: DLL load failed while importing _C: The specified module could not be found.
```

## Appendix

Full log of `pip install "git+https://github.com/facebookresearch/pytorch3d.git"`:

```
Collecting git+https://github.com/facebookresearch/pytorch3d.git
  Cloning https://github.com/facebookresearch/pytorch3d.git to e:\tmp\appdata-eliem\temp\pip-req-build-tznsqs6k
  Running command git clone -q https://github.com/facebookresearch/pytorch3d.git 'E:\tmp\AppData-eliem\Temp\pip-req-build-tznsqs6k'
Requirement already satisfied (use --upgrade to upgrade): pytorch3d==0.2.0 from git+https://github.com/facebookresearch/pytorch3d.git in e:\python38\lib\site-packages\pytorch3d-0.2.0-py3.8-win-amd64.egg
Requirement already satisfied: torchvision>=0.4 in e:\python38\lib\site-packages (from pytorch3d==0.2.0) (0.6.1)
Requirement already satisfied: fvcore in e:\python38\lib\site-packages (from pytorch3d==0.2.0) (0.1.1.post20200630)
Requirement already satisfied: torch in e:\python38\lib\site-packages (from torchvision>=0.4->pytorch3d==0.2.0) (1.5.1)
Requirement already satisfied: numpy in e:\python38\lib\site-packages (from torchvision>=0.4->pytorch3d==0.2.0) (1.19.0)
Requirement already satisfied: pillow>=4.1.1 in e:\python38\lib\site-packages (from torchvision>=0.4->pytorch3d==0.2.0) (7.2.0)
Requirement already satisfied: yacs>=0.1.6 in e:\python38\lib\site-packages (from fvcore->pytorch3d==0.2.0) (0.1.7)
Requirement already satisfied: pyyaml>=5.1 in e:\python38\lib\site-packages (from fvcore->pytorch3d==0.2.0) (5.3.1)
Requirement already satisfied: tqdm in e:\python38\lib\site-packages (from fvcore->pytorch3d==0.2.0) (4.47.0)
Requirement already satisfied: portalocker in e:\python38\lib\site-packages (from fvcore->pytorch3d==0.2.0) (1.7.0)
Requirement already satisfied: termcolor>=1.1 in e:\python38\lib\site-packages (from fvcore->pytorch3d==0.2.0) (1.1.0)
Requirement already satisfied: tabulate in e:\python38\lib\site-packages (from fvcore->pytorch3d==0.2.0) (0.8.7)
Requirement already satisfied: future in e:\python38\lib\site-packages (from torch->torchvision>=0.4->pytorch3d==0.2.0) (0.18.2)
Requirement already satisfied: pywin32!=226; platform_system == "Windows" in e:\python38\lib\site-packages (from portalocker->fvcore->pytorch3d==0.2.0) (228)
Building wheels for collected packages: pytorch3d
  Building wheel for pytorch3d (setup.py) ... done
  Created wheel for pytorch3d: filename=pytorch3d-0.2.0-cp38-cp38-win_amd64.whl size=3112729 sha256=7eee268e22362299ff8bda89419d10dd72667040c78fef33a5863bff95bcac10
  Stored in directory: E:\tmp\AppData-eliem\Temp\pip-ephem-wheel-cache-k4as0fbe\wheels\5e\a7\7c\814c3888eaaecaaf0f533faeb3c6812d82274484be51f7441c
Successfully built pytorch3d
```

**Comment by bottler** [comment]
What happens if you do `import torch` before `from pytorch3d import _C`?

**Comment by eliemichel** [comment]
Ok, there was two things, first I was linking against the wrong version of CUDA, and then indeed calling `import torch` first fixed the issue.

**Comment by eliemichel** [comment]
Actually not sure everything is installed correctly: following the first tutorial, about chamfer loss on dolphin, it seems that despite `torch.cuda.is_available()` is true it runs on CPU:

![cpu](https://user-images.githubusercontent.com/5802849/86490940-73309180-bd69-11ea-83ad-93c0b30bdd63.png)

NB: I have a single GPU so `device = torch.device("cuda:0")` should be correct.

**Comment by bottler** [comment]
I think you mean you are using the new commit of the deform meshes tutorial just now. I don't know what task manager is really showing. How quick is the main loop? Can you try restarting the kernel and setting device manually to `torch.device("cpu")` instead and see if it gets slower?

**Comment by eliemichel** [comment]
Oh yeah forgot to mention I am running everything local (jupyter is on http://localhost) so the task manager is the one of the machine running the ipython kernel. It takes 1min 56s with cuda and 12 min with a cpu tensor so there is a difference still.

Note that the previous screenshot was taken on an example with more points (an icosphere with n=5), with the exact tutorial code the CPU load is more around 50% (but well, it depends on the chip anyway), and the GPU load is even less, like 1 or 2 percents... feels underused. ^^

**Comment by weizmanl** [comment]
Hi, 

Has anyone solved this issue for windows 10? I did everything as requested and I still get: "cannot import name '_C' from 'pytorch3d'".

I also see that there is a # pyre-fixme[21]: Could not find name `_C` in `pytorch3d`. comment.

Is this a known issue?

Thank you

**Comment by bottler** [comment]
@weizmanl Yes, pytorch3d can be built on Windows. But it's quite fiddly. I suggest you set up a new conda environment with pytorch 1.6.0 and make the changes to the pytorch sources given in issue #323. If you have problems, please open a new issue.

**Comment by bottler** [comment]
A performance improvement to the mesh_normal_consistency calculation has just landed (4bfe7158b) which you should see makes a big difference to this experience. I am therefore closing this issue.
